.github/workflows/build-and-deploy.yml

```diff
@@ -83,25 +84,36 @@ jobs:
     - name: Verify Build
       run: |
         file "${{ env.OUTPUT }}"
-        openssl dgst -sha256 "${{ env.OUTPUT }}" | tee bettercap_${{ matrix.target_os }}_${{ matrix.target_arch }}_${{ env.VERSION }}.sha256
-        7z a "bettercap_${{ matrix.target_os }}_${{ matrix.target_arch }}_${{ env.VERSION }}.zip" "${{ env.OUTPUT }}" "bettercap_${{ matrix.target_os }}_${{ matrix.target_arch }}_${{ env.VERSION }}.sha256"
+        openssl dgst -sha256 "${{ env.OUTPUT }}" | tee bettercap_${{ matrix.os.name }}_${{ matrix.arch }}.sha256
+        7z a "bettercap_${{ matrix.os.name }}_${{ matrix.arch }}.zip" "${{ env.OUTPUT }}" "bettercap_${{ matrix.os.name }}_${{ matrix.arch }}.sha256"
+
+    - name: Upload Artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: release-artifacts-${{ matrix.os.name }}-${{ matrix.arch }}
+        path: |
+          bettercap_*.zip
+          bettercap_*.sha256
 
   deploy:
     needs: [build]
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')
```

Comment:
Note: this check has moved to the "Upload Release Assets" step below in order to be able to list artifacts and check everyhing is OK. But it's the same behavior: nothing will be published until a new tag is created.